data/transactions/logic/chacha20rng.go

```diff
@@ -0,0 +1,45 @@
+package logic
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"gitlab.com/yawning/chacha20.git"
```

Comment:
go.mod has a commit hash version recorded. author hasn't tag-versioned anything. I'm slightly tempted to fork their project just to add tag versions. Or we could dump this and use a C wrapper around the libsodium chacha20 implementation since we already link in libsodium.

Reply:
I agree that we should do one of those two options